src/utils/ucc_log.h

```diff
@@ -115,10 +116,6 @@ static inline const char* ucc_datatype_str(ucc_datatype_t dt)
         return "int128";
     case UCC_DT_UINT128:
         return "uint128";
-    case UCC_DT_USERDEFINED:
```

Comment:
we still need to return some str in case of userdefined for example to print it in ucc_coll_str, add
```
if (!UCC_DT_IS_PREDEFINED(dt)) {
    return "userdefined";
}
```